### ISSUE TITLE
Add MCTS-based trained bot personality

### ISSUE THREAD

**Issue by perahrens** [perahrens/baisch#192]
## Background

The current bot personalities (Passive, Balanced, Aggressive, Tactician) use hand-crafted heuristics. A Monte Carlo Tree Search (MCTS) bot would instead **explore possible futures** to pick the best action — no external API, no training data required, just the game rules.

## Concept

MCTS works by repeatedly simulating random playouts from the current game state and backing up win/loss results to guide action selection. For Baisch this means:

1. **Expand**: enumerate legal actions for the current bot (plunder deck A/B, attack each face-up defense slot, king attack if available, pass/expose).
2. **Simulate**: from each candidate action, run a fast random rollout (bots play randomly) to end-of-game.
3. **Backpropagate**: update win counts for each visited action node.
4. **Select**: choose the action with the highest win rate (UCB1 formula balances exploitation vs. exploration).

After enough iterations within a time budget (e.g., 200 ms), the action with the most simulated wins is executed.

## Why MCTS fits Baisch

- The game is **fully observable** for the bot (server-side, all state is known).
- Turn count is bounded (~20-40 turns), so playouts terminate quickly.
- No training pipeline needed — just implement the rules in a fast simulator.
- Naturally handles complex multi-card combos and hero interactions.

### 1. Fast game simulator (`server/mcts-sim.js`)
A stripped-down, synchronous copy of `gameState.js` that supports:
- `clone()` — deep-copy current state
- `legalActions(playerIdx)` — returns array of action objects
- `applyAction(action)` — mutates state, advances turn
- `winner()` — returns winning player index or -1

This simulator does **not** emit socket events or touch `io`.

### 2. MCTS engine
```js
function mctsSearch(simState, playerIdx, iterationCount) {
  // Build tree, run iterationCount simulations, return best action
}
```
Uses UCB1: score = wins/visits + C * sqrt(ln(parentVisits) / visits)

### 3. New personality: `MctsPersonality`
- Registered as mode key `"mcts"` / display label `"MCTS"`
- In `executeBotTurn`: calls `mctsSearch(gs.clone(), idx, 300)` synchronously (fast enough for Node single-threaded event loop at ~300 iterations)
- Falls back to Tactician if MCTS returns null

### 4. Self-play data logging (optional, future)
Once MCTS is working, add optional game log export (`--log-games`) to generate training data for a future neural net policy.

## Acceptance Criteria

- [ ] `mcts-sim.js` passes a unit test: simulate a 2-player game to completion
- [ ] MCTS bot can be selected from the game lobby (same UI as other personalities)
- [ ] MCTS bot always makes a legal move; never throws or hangs
- [ ] Bot responds within 2 seconds (MCTS iteration count tuned to fit)
- [ ] Win rate vs. Tactician bot ≥ 50% in 100 simulated games
